src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,17 +25,30 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
+        value = address;
```

Comment:
if you retain this, you will be introducing inconsistency due to duplication (please recollect the discussion around this during the tutorial)
the idea is to store the address info in 4 parts and remove this..